logging/google/cloud/logging/handlers/app_engine.py

```diff
@@ -29,45 +29,55 @@
+        'module_id': os.getenv('GAE_SERVICE'),
+        'version_id': os.getenv('GAE_VERSION'),
+    },
+)
 
-    Writes to the file that the fluentd agent on App Engine Flexible is
-    configured to discover logs and send them to  Stackdriver Logging.
-    Log entries are wrapped in JSON and with appropriate metadata. The
-    process of converting the user's formatted logs into a JSON payload for
-    Stackdriver Logging consumption is implemented as part of the handler
-    itself, and not as a formatting step, so as not to interfere with
-    user-defined logging formats.
+
+class AppEngineHandler(CloudLoggingHandler):
+    """A handler that directly makes Stackdriver logging API calls.
+
+    This handler can be used to route Python standard logging messages directly
+    to the Stackdriver Logging API.
+
+    This handler supports both an asynchronous and synchronous transport.
+
+    :type client: :class:`google.cloud.logging.client`
+    :param client: the authenticated Google Cloud Logging client for this
+                   handler to use
+
+    :type name: str
+    :param name: the name of the custom log in Stackdriver Logging. Defaults
+                 to 'python'. The name of the Python logger will be represented
+                 in the ``python_logger`` field.
+
+    :type transport: type
+    :param transport: Class for creating new transport objects. It should
+                      extend from the base :class:`.Transport` type and
+                      implement :meth`.Transport.send`. Defaults to
+                      :class:`.BackgroundThreadTransport`. The other
+                      option is :class:`.SyncTransport`.
+
+    :type resource: :class:`~google.cloud.logging.resource.Resource`
+    :param resource: Monitored resource of the entry, defaults
+                     to the global resource type.
     """
 
-    def __init__(self):
-        """Construct the handler
-
-        Large log entries will get mangled if multiple workers write to the
-        same file simultaneously, so we'll use the worker's PID to pick a log
-        filename.
-        """
-        self.filename = _LOG_PATH_TEMPLATE.format(pid=os.getpid())
-        super(AppEngineHandler, self).__init__(self.filename,
-                                               maxBytes=_MAX_LOG_BYTES,
-                                               backupCount=_LOG_FILE_COUNT)
-
-    def format(self, record):
-        """Format the specified record into the expected JSON structure.
-
-        :type record: :class:`~logging.LogRecord`
-        :param record: the log record
-
-        :rtype: str
-        :returns: JSON str to be written to the log file
-        """
-        message = super(AppEngineHandler, self).format(record)
-        return format_stackdriver_json(record, message)
+    def __init__(self, client,
+                 name=DEFAULT_LOGGER_NAME,
+                 transport=BackgroundThreadTransport,
+                 resource=GAE_RESOURCE):
```

Comment:
Since we never expect users to be able to change the resource, I would remove this from the constructor.